tooling/unpack-macros.mjs

```diff
@@ -2,47 +2,106 @@
  */
 
-import * as fs from "fs";
-import * as path from "path";
+import fs from "fs";
+import path from "path";
+
+import {extractPack} from "@foundryvtt/foundryvtt-cli";
+import chalk from "chalk";
+
 import {DIR_PACKS, DIR_PACKS_SOURCE} from "./consts.mjs";
+import {getSafeFilename} from "./utils.js";
+import {isMacroKey} from "./utils-pack.js";
+
+const _UNWANTED_DOC_KEYS = [
+	"author",
+	"ownership",
+	"flags",
+	"_stats",
+];
+
+// Discover pack dirs
+const packDirs = fs.readdirSync(DIR_PACKS)
+	.map(name => ({
+		name,
+		dirpath: path.join(DIR_PACKS, name),
+	}))
+	.filter(({dirpath}) => fs.statSync(dirpath).isDirectory());
+
+// Unpack packs
+for (const {name, dirpath} of packDirs) {
+	const dirpathOut = path.join(DIR_PACKS_SOURCE, name);
+
+	console.log(`Unpacking "${name}"`);
+	console.group();
+
+	const entryToFolder = {};
+	const folders = {};
+
+	await extractPack(
+		dirpath,
+		dirpathOut,
+		{
+			clean: true,
+			log: true,
+			transformEntry: (entry) => {
+				entryToFolder[entry._id] = entry.folder;
+
+				_UNWANTED_DOC_KEYS.forEach(key => delete entry[key]);
+
+				if (entry._key.startsWith("!folders!")) {
+					folders[entry._id] = {
+						name: entry.name,
+						folder: entry.folder,
+					};
+					return;
+				}
+
+				if (!isMacroKey(entry._key)) return;
+
+				// For macro entries, write the `"command"` out as a separate `.js` file
+				const scriptFilename = `${getSafeFilename(`${entry.name}.${entry._id}`)}.js`;
+				fs.writeFileSync(path.join(dirpathOut, scriptFilename), entry.command, "utf-8");
+				console.log(`Wrote ${chalk.cyan(scriptFilename)}`);
+				delete entry.command;
+			},
+		},
+	);
+
+	// Folderize files
+	const getDirPathParts = (json, stack) => {
+		stack ||= [];
+		if (!json.folder) return stack;
+		stack.unshift(folders[json.folder].name);
+		return getDirPathParts(folders[json.folder], stack);
+	};
+
+	fs.readdirSync(dirpathOut)
+		.filter(name => name.endsWith(".json"))
+		.forEach(name => {
+			const fpath = path.join(dirpathOut, name);
+			const json = JSON.parse(fs.readFileSync(fpath, "utf-8"));
+
+			const dirPathParts = getDirPathParts(json);
+
+			// Skip documents which are not in a folder, and are not a macro
+			if (!dirPathParts.length && !isMacroKey(json._key)) return;
+
+			const dirPath = path.join(dirpathOut, ...dirPathParts);
+			fs.mkdirSync(dirPath, {recursive: true});
+			const fpathFolderized = path.join(dirPath, name);
+
+			// For non-macro documents, just move the JSON
+			if (!isMacroKey(json._key)) {
+				fs.renameSync(fpath, fpathFolderized);
+				return;
+			}
+
+			// For macro documents, track the `"command"` source, and write out the modified file
+			json.command_source = fpathFolderized.replace(/\.json$/, ".js").split(path.sep).join("/");
+			fs.writeFileSync(fpathFolderized, JSON.stringify(json, null, 2) + "\n", "utf-8");
+			fs.renameSync(fpath.replace(/\.json$/, ".js"), json.command_source);
+			if (dirPathParts.length) fs.unlinkSync(fpath);
+		});
```

Comment:
Normally, unpacking a v11+ compendium using `@foundryvtt/foundryvtt-cli` dumps it as one giant flat directory with no folders. I do not like this, as it is not human-friendly. Therefore, I have hacked in some changes to move each file to a directory named the same as its in-compendium folder. 

For example, if I have the following structure in a Foundry compendium:

```
Effects/
  Gun 1
Utilities/
  Helpful Macro
```

And run a default unpack, I would get the following structure on disk:

```
Effects.json
Gun 1.json
Helpful Macro.json
Utilities.json
```

With the changes here, we instead get:

```
Effects.json
Effects/
  Gun 1.json
Utilities.json
Utilities/
  Helpful Macro.json
```

When there are a lot of macros, like we have in LWFX, this makes it much easier to navigate!

---

Unfortunately, this inefficient in that lots of files get read and/or written twice. The alternatives were:

- Make my own version of `@foundryvtt/foundryvtt-cli` -- bad
- Make a PR to `@foundryvtt/foundryvtt-cli` to add the functionality I want -- going from simple "flat files" to "nested directory tree" has a lot of complexity and edge cases, which I am (reasonably) confident I have handled for our limited use-case, but I lack the energy to generalize it for the CLI (and convince the maintainers that merging it is a good idea). So not feasible.
- Leave it as-is and use the ugly flat files -- bad
- This! -- *okay*

It does not add any perceivable slowdown on my laptop with average CPU and SSD so I am happy with it for now. In future I can ask the CLI maintainers if they would accept such a PR.